packages/playground/remote/src/lib/boot-playground-remote.ts

```diff
@@ -129,8 +129,35 @@ export async function bootPlaygroundRemote() {
 			// Manage the address bar
 			wpFrame.addEventListener('load', async (e: any) => {
 				try {
+					/**
+					 * When navigating to a page with %0A sequences (encoded newlines)
+					 * in the query string, the `location.href` property of the
+					 * iframe's content window doesn't seem to reflect them. Everything
+					 * else is in place, but not the %0A sequences.
+					 *
+					 * Weirdly, these sequences are available after the next event
+					 * loop tick – hence the `setTimeout(0)`.
+					 *
+					 * The exact cause is unclear at the moment of writing of this
+					 * comment. The WHATWG HTML Standard [1] has a few hints:
+					 *
+					 * * Current and active session history entries may get out of
+					 *   sync for iframes.
+					 * * Documents inside iframes have "is delaying load events" set
+					 *   to true.
+					 *
+					 * But there doesn't seem to be any concrete explanation and no
+					 * recommended remediation. If anyone has a clue, please share it
+					 * in a GitHub issue or start a new PR.
+					 *
+					 * [1] https://html.spec.whatwg.org/multipage/document-sequences.html#nav-active-history-entry
+					 */
+					// Get the content window while e.currentTarget is available.
+					// It will be undefined on the next event loop tick.
+					const contentWindow = e.currentTarget!.contentWindow;
```

Comment:
Could this be related to the async event handler?

I've done some testing (not on playground) and with an async event handler the behavior seemed less reliable. However, with the sync event handler, `e.target.contentWindow.location.href` seemed to reliably be the expected value.

I don't know whether `target`/`currentTarget` matters, I suspect not, but `target` seemed to work well in my testing the load event on the iframe.

Reply:
I tried replacing the `async` code with a callback but it didn't work.

Reply:
Yeah, the information is lost after the `await`. I was thinking it's related to React and how it clears `currentTarget` after the synchronous event handler returns, but now that I think about it, React shouldn't be in the mix here.